src/lib.rs

```diff
@@ -1080,6 +1129,17 @@ mod tests {
             )
             .unwrap();
     }
+
+    fn with_default_system_tags<T, F: FnOnce() -> T>(f: F) -> T {
+        std::env::set_var("DD_ENV", "production");
+        std::env::set_var("DD_SERVICE", "service");
+        std::env::set_var("DD_VERSION", "0.0.1");
+        let t = f();
+        std::env::remove_var("DD_ENV");
+        std::env::remove_var("DD_SERVICE");
+        std::env::remove_var("DD_VERSION");
+        t
+    }
```

Comment:
This causes flakiness. I've had `test_new_default_tags` randomly fail locally because of this.

Reply:
Can i propose to put in the readme something like a `test` section suggesting to test using the `--test-threads 1`?